### DISCUSSION TITLE
Odometry: in 24.4.8.6

### DISCUSSION THREAD

**Discussion by VikingNation** [BroncBotz3481/YAGSL-Example#204] [Q&A] [answered]
I have a question about how odometry in YAGSL library version 24.4.8.6 is being reported.

When using Shuffleboard dashboard when we press forward on our controller the robot moves forward in the real world which is correct.  On Shuffleboard the Robot moves backwards; it moves from right to left on the display.   This behavior is the same when running code on our robot and simulating the code.  Switching from Red to Blue Aliance in the Driver Station and Simulator does not change the behavior.  This behavior is also the same in the most recent commit of the YAGSL project (8c3470f453451bb0d2bbef8f6ef6018b8311b9ea).   

In field-oriented driving when pressing forward the robot should move away from the driver station across the game field. Shouldn't the odometry be increasing when moving away from the driver station?

**Comment by thenetworkgrinch** [comment]
How does it look in FRC Web Components? 

If it does the same thing, try adding 180 to every module absolute encoder offset, you may have to run through the 8 steps again but this usually works.

**Reply by VikingNation** [reply, marked as answer]
We resolved the issue.  The Xbox controller follows a South-East convention (positive down and right).  Added -1 times joystick input to align to a North-West convention.  Then inverted drive motor (already had turn motor inverted) in JSON files for all modules.  The issue resolved. Odometry increases when going forward and strafe left.

**Reply by thenetworkgrinch** [reply]
Awesome! Glad you found it!
